angular-swfobject.js

```diff
@@ -46,23 +46,37 @@ angular.module('swfobject', [])
           });
         }
 
-
         // http://learnswfobject.com/advanced-topics/executing-javascript-when-the-swf-has-finished-loading/
         function swfLoadEvent(evt, fn) {
-            //Set up a timer to periodically check value of PercentLoaded
-            var loadCheckInterval = $interval(function () {
-              //Ensure Flash Player's PercentLoaded method is available and returns a value
-              if (typeof evt.ref.PercentLoaded !== "undefined" && evt.ref.PercentLoaded()) {
-                //Once value == 100 (fully loaded) we can do whatever we want
-                if (evt.ref.PercentLoaded() === 100) {
-                  //Clear interval
-                  $interval.cancel(loadCheckInterval);
-                  loadCheckInterval = null;
-                  //Execute function
-                  fn({evt: evt});
+            //This interval ensures we don't try to access PercentLoaded too soon
+            var startCheckInterval = $interval(function () {
+                //Ensure Flash Player's PercentLoaded method is available
+                if (typeof evt.ref.PercentLoaded !== 'undefined') {
+                    $interval.cancel(startCheckInterval);
+                    startCheckInterval = null;
```

Comment:
The problem with time outs is that they are only ever going to be called
once. So you have to set generous initial wait times to compensate.

If you are loading many small flash files (my use case is that of unloading
one and loading the next in quick succession) then the lag makes for poor
UX. Better to use small intervals instead.
On 14 Nov 2015 07:23, "Jeff Knaggs" <notifications@github.com> wrote:

> In angular-swfobject.js
> <https://github.com/jeef3/angular-swfobject/pull/8#discussion_r44829669>:
>
> > -            var loadCheckInterval = $interval(function () {
> > -              //Ensure Flash Player's PercentLoaded method is available and returns a value
> > -              if (typeof evt.ref.PercentLoaded !== "undefined" && evt.ref.PercentLoaded()) {
> > -                //Once value == 100 (fully loaded) we can do whatever we want
> > -                if (evt.ref.PercentLoaded() === 100) {
> > -                  //Clear interval
> > -                  $interval.cancel(loadCheckInterval);
> > -                  loadCheckInterval = null;
> > -                  //Execute function
> > -                  fn({evt: evt});
> > +            //This interval ensures we don't try to access PercentLoaded too soon
> > +            var startCheckInterval = $interval(function () {
> > +                //Ensure Flash Player's PercentLoaded method is available
> > +                if (typeof evt.ref.PercentLoaded !== 'undefined') {
> > +                    $interval.cancel(startCheckInterval);
> > +                    startCheckInterval = null;
>
> We could probably avoid all this interval checking and cancelling by using
> timeouts instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jeef3/angular-swfobject/pull/8/files#r44829669>.
>